sdks/python/apache_beam/runners/interactive/background_caching_job.py

```diff
@@ -231,9 +231,21 @@ def has_source_to_cache(user_pipeline):
                       streaming_cache.StreamingCache):
 
       file_based_cm = ie.current_env().get_cache_manager(user_pipeline)
+      cache_dir = file_based_cm._cache_dir
+      from apache_beam.runners.interactive import interactive_beam as ib
+      if ib.options.cache_root:
+        _LOGGER.warning(
+            'Interactive Beam has detected a set value for the cache_root '
+            'option. Please note: existing cache managers will not have '
```

Comment:
You may move the warning to the setter of the property in interactive_beam module. Then you only need to code this logging once when the value is modified.